File: RCSceneExample-OC/RCSceneExample-OC/Feature/Login/ViewController/LoginViewController.m
//
//  LoginViewController.m
//  RCSceneExample-OC
//
//  Created by shaoshuai on 2022/5/6.
//

#import "LoginViewController.h"

#import "RCSceneBridge.h"

@interface LoginViewController ()

@end

@implementation LoginViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    // Do any additional setup after loading the view.
    self.view.backgroundColor = UIColor.whiteColor;
}

- (void)viewWillAppear:(BOOL)animated {
    [super viewWillAppear:animated];
    
    [self.textField becomeFirstResponder];
}

- (IBAction)login {
    [self.view endEditing:YES];
    
    NSString *phone = self.textField.text;
    if (phone.length == 0) {
        return;
        //        return SVProgressHUD.showError(withStatus: "请输入正确手机号")
    }
    
    __weak typeof(self) weakSelf = self;
    [RCSceneBridge login:phone success:^(UserModel * model) {
        [weakSelf.navigationController popViewControllerAnimated:YES];
    } failure:^(NSString * error) {
        NSLog(@"login error %@", error);
    }];
}

@end

File: RCSceneExample/RCSceneExample/LoginViewController.swift
//
//  LoginViewController.swift
//  RCSceneExample
//
//  Created by shaoshuai on 2022/3/29.
//

import UIKit
import SVProgressHUD

class LoginViewController: UIViewController {
    
    @IBOutlet weak var phoneTF: UITextField!
    
    private lazy var service = RCVideoRoomService()

    override func viewDidLoad() {
        super.viewDidLoad()

        // Do any additional setup after loading the view.
    }
    
    override func viewWillAppear(_ animated: Bool) {
        super.viewWillAppear(animated)
        phoneTF.becomeFirstResponder()
    }
    
    @IBAction func login() {
        view.endEditing(true)
        
        guard let text = phoneTF.text, text.count == 11 else {
            return SVProgressHUD.showError(withStatus: "请输入正确手机号")
        }
        
        service.login(phone: text) { [weak self] result in
            switch result {
            case .success:
                self?.navigationController?.popViewController(animated: true)
            case let .failure(error):
                SVProgressHUD.showError(withStatus: error.localizedDescription)
            }
        }
    }

}
